src/client/utils/download/download.ts

```diff
@@ -51,11 +58,34 @@ export function datasetToFileString(dataset: Dataset, fileFormat: FileFormat, op
   }
 }
 
-export function makeFileName(...args: string[]): string {
-  var nameComponents: string[] = [];
-  args.forEach(arg => {
-    if (arg) nameComponents.push(arg.toLowerCase());
-  });
-  var nameString = nameComponents.join("_");
-  return nameString.length < 200 ? nameString : nameString.substr(0, 200);
+function dateToFileString(date: Date): string {
+  return date.toISOString()
+    .replace("T", "_")
+    .replace("Z", "")
+    .replace(".000", "");
+}
+
+export function filter2NameComponent(filter: Filter): string {
+  const timeFilter: FixedTimeFilterClause = filter.clauses.find(clause => clause instanceof FixedTimeFilterClause) as FixedTimeFilterClause;
+  const nonTimeClauseSize = filter.clauses.filter(clause => !(clause instanceof FixedTimeFilterClause)).count();
+  const filtersPart = nonTimeClauseSize === 0 ? "" : `_filters-${nonTimeClauseSize}`;
+  if (timeFilter) {
+    const { start, end } = timeFilter.values.first();
+    return `${dateToFileString(start)}_${dateToFileString(end)}${filtersPart}`;
+  }
+  return filtersPart;
+}
+
+const split2NameComponent = (split: Split) => `${STRINGS.splitDelimiter}_${split.reference}`;
```

Comment:
As above, I'm not sure that split definition should be a part of downloaded file name.